### PR TITLE
tests: Add base test for C++ exceptions handling.

### PR DESCRIPTION
**Pull request by r2r0**
Try to throw and catch C++ exception of int type.

Related to: #32448 and #34229

Signed-off-by: Artur Lipowski <Artur.Lipowski@hidglobal.com>

### CODE REVIEW

**Review by galak — CHANGES_REQUESTED** [review]
This should get added to tests/application_development/libcxx

**Review by galak — COMMENTED** [review]
Looks general good, slightly structure change:

```
#ifdef CONFIG_EXCEPTIONS
static void throw_exception(void)
{
	throw 42;
}

static void test_exception(void)
{
   try...
}
#else 

static void test_exception(void)
{
   ztest_test_skip()
}
#endif 
```

File: tests/application_development/libcxx/src/main.cpp [hunk 1]
```diff
@@ -75,6 +97,12 @@ void test_main(void)
 			 ztest_unit_test(test_vector),
 			 ztest_unit_test(test_make_unique)
 		);
```
Comment (galak) [inline tests/application_development/libcxx/src/main.cpp:99]: Just add `ztest_unit_test(test_exception)`

(and ifdef w/`ztest_test_skip()`)

**Review by galak — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by nashif** [comment]
hmm, we know this is broken (#32448 and #34229), now those tests will be failing on all those platforms

**Pull request merged**